api/lib/opentelemetry/distributed_context/entry.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module DistributedContext
+    # An Entry consists of Entry::Metadata, Entry::Key, and Entry::Value.
+    class Entry
+      attr_reader :metadata, :key, :value
+
+      # Entry::Key is the name of the Entry. Entry::Key along with Entry::Value can be used to aggregate and group stats,
+      # annotate traces and logs, etc.
+      #
+      # Restrictions
+      # - Must contain only printable ASCII (codes between 32 and 126 inclusive)
+      # - Must have length greater than zero and less than 256.
+      # - Must not be empty.
+      class Key
+        attr_reader :name
+
+        def initialize(name)
+          # TODO: name should be printable ascii (32..126), not just ascii.
+          raise ArgumentError unless name&.ascii_compatible? && (1..255).include?(name.length)
+
+          @name = -name
```

Comment:
Why `-name`?

Reply:
We want this to be a frozen copy. The spec requires it to be immutable.

Reply:
I was just surprised because I haven't seen this syntax before and cannot find and documentation about it.

Reply:
https://ruby-doc.org/core-2.4.0/String.html#method-i-2D-40